### PR TITLE
use directory-item-selector from commons-ui

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -407,6 +407,33 @@ export function fetchRootFolders(types) {
     return backendFetchJson(fetchRootFoldersUrl);
 }
 
+export function fetchElementsMetadata(ids, elementTypes, equipmentTypes) {
```
Comment (FranckLecuyer) [inline src/utils/rest-api.js]: Why not using existing fetchElementsInfos function ? 
Here, there is just an additional parameter equipmentTypes to deal with, but the endpoint called in explore-server is the same ...

File: src/components/dialogs/move-dialog.js [hunk 2]
```diff
@@ -28,7 +34,12 @@ const MoveDialog = ({ open, onClose, items }) => {
                     nbElements: items.length,
                 }
             )}
-            contentText={intl.formatMessage({ id: 'moveItemContentText' })}
```
Comment (FranckLecuyer) [inline src/components/dialogs/move-dialog.js:31]: Why removing this property contentText ?

File: src/components/dialogs/commons/modify-element-selection.tsx [hunk 3]
```diff
@@ -6,27 +6,31 @@
...
+    fetchRootFolders,
+} from '../../../utils/rest-api';
 
-export interface DirectorySelectProps {
-    types: ElementType[];
+export interface ModifyElementSelectionProps {
+    elementType: ElementType;
     dialogOpeningButtonLabel: string;
     dialogTitleLabel: string;
-    dialogMessageLabel: string;
```
Comment (FranckLecuyer) [inline src/components/dialogs/commons/modify-element-selection.tsx:22]: Why removing dialogMessageLabel property ?

File: src/components/dialogs/commons/modify-element-selection.tsx [hunk 4]
```diff
@@ -98,22 +102,24 @@ const DirectorySelect: React.FunctionComponent<DirectorySelectProps> = (
...
-                title={intl.formatMessage({
-                    id: props.dialogTitleLabel,
-                })}
+                types={[props.elementType]}
+                onlyLeaves={props.elementType !== ElementType.DIRECTORY}
+                multiselect={false}
                 validationButtonText={intl.formatMessage({
                     id: 'confirmDirectoryDialog',
                 })}
-                contentText={intl.formatMessage({
```
Comment (FranckLecuyer) [inline src/components/dialogs/commons/modify-element-selection.tsx:111]: Why removing this property contentText ?

File: src/components/dialogs/create-study-dialog/create-study-dialog.js [hunk 5]
```diff
@@ -269,11 +269,10 @@ const CreateStudyDialog = ({ open, onClose, providedExistingCase }) => {
                 </Grid>
             </Grid>
             {providedExistingCase ? (
-                <DirectorySelect
-                    types={[ElementType.DIRECTORY]}
+                <ModifyElementSelection
+                    elementType={ElementType.DIRECTORY}
                     dialogOpeningButtonLabel={'showSelectDirectoryDialog'}
                     dialogTitleLabel={'selectDirectoryDialogTitle'}
-                    dialogMessageLabel={'moveItemContentText'}
```
Comment (FranckLecuyer) [inline src/components/dialogs/create-study-dialog/create-study-dialog.js:276]: Why removing dialogMessageLabel property ?

File: src/components/dialogs/filter/explicit-naming/explicit-naming-filter-form.tsx [hunk 6]
```diff
@@ -240,12 +240,11 @@ export const ExplicitNamingFilterForm: FunctionComponent<
...
                 />
                 {props.sourceFilterForExplicitNamingConversion && (
-                    <DirectorySelect
-                        types={[ElementType.STUDY]}
+                    <ModifyElementSelection
+                        elementType={ElementType.STUDY}
                         onElementValidated={onStudySelected}
                         dialogOpeningButtonLabel={'selectStudyDialogButton'}
                         dialogTitleLabel={'selectStudyDialogTitle'}
-                        dialogMessageLabel={'selectStudyText'}
```
Comment (FranckLecuyer) [inline src/components/dialogs/filter/explicit-naming/explicit-naming-filter-form.tsx]: Why removing dialogMessageLabel property ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/components/dialogs/move-dialog.js (hunk 2 above)
Comment (dbraquart) [reply src/components/dialogs/move-dialog.js:31]: oops I though you removed this prop. I will restore everywhere.

File: src/utils/rest-api.js (hunk 1 above)
Comment (dbraquart) [reply src/utils/rest-api.js]: yes. good catch

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**